tests/CMakeLists.txt

```diff
@@ -8,13 +8,21 @@ foreach(test ${tests_opentime})
 	add_executable(${test} utils.h utils.cpp ${test}.cpp)
 	target_link_libraries(${test} opentime)
 	set_target_properties(${test} PROPERTIES FOLDER tests)
-	add_test(${test} ${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/${test}${CMAKE_EXECUTABLE_SUFFIX})
+	add_test(NAME ${test} 
+           COMMAND ${test}
+           # Set the pwd to the source directory so we can load the samples
+           # like the python tests do
+           WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/sample_data)
```

Comment:
@darbyjohnston just want to make sure you're okay with some of the changes I made to the cpp tests.

Reply:
The extra assert functions look good and I like the change for printing out the test names.

The ```WORKING_DIRECTORY``` also seems like a good change if that's how the Python tests work. The only concern I have is that if a test writes out a temporary file to the current directory it would end up in the source tree, but I guess that's not an issue if the Python tests aren't doing that.

Reply:
That's true, good point. The convention I am used to for unit test output is to write to the system temp directory, and that seems to be what the python tests are doing as well.  As long as the c++ tests follow the same convention we should be alright.

Reply:
`WORKING_DIRECTORY` seems good to me as long as there's no files written to it.